Shop_Unity_base/Assets/Scripts/Shop/Model/ShopModel.cs

```diff
@@ -2,166 +2,182 @@
-        //returns the number of items
+        // Returns the number of items
         public int GetItemCount()
         {
             return itemList.Count;
         }
 
-        //------------------------------------------------------------------------------------------------------------------------
-        //                                                  GetItemByIndex()
-        //------------------------------------------------------------------------------------------------------------------------        
-        //tries to get an item, specified by index. returns null if unsuccessful
+        // Rries to get an item, specified by index. returns null if unsuccessful
         public Item GetItemByIndex(int index)
         {
             if (index >= 0 && index < itemList.Count)
-            {
                 return itemList[index];
-            }
             else
-            {
                 return null;
-            }
         }
 
-        //------------------------------------------------------------------------------------------------------------------------
-        //                                                  GetMessage()
-        //------------------------------------------------------------------------------------------------------------------------        
-        //returns the cached list of messages
+        // Returns the cached list of messages
         public string[] GetMessages()
         {
-            return messages.ToArray();
+            // TODO this defeats the purpose of a queue
+            string[] queuedMessages = messages.ToArray();
+            messages.Clear();
+
+            return queuedMessages;
         }
 
-        //------------------------------------------------------------------------------------------------------------------------
-        //                                                  AddMessage()
-        //------------------------------------------------------------------------------------------------------------------------
-        //adds a message to the cache, cleaning it up if the limit is exceeded
+        // Adds a message to the cache, cleaning it up if the limit is exceeded
         private void AddMessage(string message)
         {
-            messages.Add(message);
-            while (messages.Count > MaxMessageQueueCount)
+            messages.Enqueue(message);
+
+            while (messages.Count > MAX_MESSAGES_IN_QUEUE)
+                messages.Dequeue();
+        }
+
+        public void SetCustomer(Customer customer)
+        {
+            this.customer = customer;
+        }
+
+        public Item Buy()
+        {
+            Item item = GetSelectedItem();
+
+            if (customer.Gold < item.Cost)
+            {
+                AddMessage("You do not have enough gold to buy that item!");
+                AddMessage("Item cost: " + item.Cost + ", Your gold: " + customer.Gold);
```

Comment:
You could use advanced C# string formatting, i.e.
```suggestion
                AddMessage($"Item cost: {item.Cost}, Your gold: {customer.Gold}");
```

Reply:
Which you apparently do use later on... Huh.

Reply:
I've only started using that recently, and I forget, thanks for pointing it out there :)